apps/flame-defi/app/features/EvmWallet/contexts/EvmWalletContext.tsx

```diff
@@ -287,6 +306,91 @@ export const EvmWalletProvider: React.FC<EvmWalletProviderProps> = ({
     resetState();
   }, [disconnect, resetState]);
 
+  const approveToken = useCallback(
+    async (token: EvmCurrency | null | undefined) => {
+      if (
+        !userAccount.address ||
+        !walletClient ||
+        !contracts?.swapRouter?.address
+      ) {
+        return null;
+      }
+      const amountAsBigInt = BigInt(defaultApprovalAmount);
+      // NOTE: Reset this to 0 whenever we want to reset the approval
+      // const amountAsBigInt = BigInt('0');
+
+      try {
+        const txHash = await walletClient.writeContract({
```

Comment:
ERC20 interactions should be implemented via a new class that extends `GenericContractService`